pkgs/tools/nix/nixos-render-docs/README.md

```diff
@@ -0,0 +1,82 @@
+# nixos-render-docs
+
+A [CommonMark](https://commonmark.org/) and [`man-pages`](https://www.man7.org/linux/man-pages/man7/man-pages.7.html) renderer for the NixOS and Nixpkgs manuals.
+
+## Summary
+
+NixOS Render Docs is a renderer originally made for documentation written in the [DocBook](https://docbook.org/whatis) format, which was then dropped in support for CommonMark, a flavour of markdown, following [this RFC](https://github.com/NixOS/rfcs/pull/72). This shift aligns the project with more accessible documentation formats and community standards (along with the recent shift to [Diátaxis](https://diataxis.fr/)), enabling easier contributions and enhancing readability for new users.
+
+Having our own renderer gives us a few key benefits:
+- **Control and Flexibility**: We own the code, we can manipulate and extend it however we like. An example of this is the integration of option docs and our solution to redirecting away from broken links, as described in detail below.
+- **Scalability and Extensibility**: As the NixOS documentation grows, having a renderer under our control makes it easier to scale and introduce new features without being limited by the constraints of external tools. Future enhancements, such as tested code blocks or localized versions of documentation, can be seamlessly integrated.
```

Comment:
```suggestion
Maintaining our own documentation rendering framework may appear extreme but has practical advantages:
- We never have to work around existing tools made under different assumptions
- We don't have to deal with unexpected breakage
- We can grow the framework with our evolving requirements without relying on external support or approval or the need to maintain a small diff to upstream
- The amount of code involved is minimal because it's single-purpose
```